scripts/ci/assay_runner_delegated_proof_pack.py

```
@@ -246,8 +436,29 @@ def self_test() -> None:
         proof_root = root / "proof"
+        ebpf_object = root / "target" / "assay-ebpf.o"
+        ebpf_object.parent.mkdir(parents=True)
+        ebpf_object.write_bytes(b"ebpf")
+        if validate_path_within_root(Path("target/assay-ebpf.o"), root, label="self-test") != ebpf_object.resolve(
+            strict=False
+        ):
+            raise ProofPackError("self-test relative path did not resolve under workspace root")
         ebpf_provenance = root / "assay-ebpf.provenance.json"
-        ebpf_provenance.write_text('{"schema":"assay.ci.ebpf_build_provenance.v0"}\n', encoding="utf-8")
+        path_trees = {
+            path: {"oid": hashlib.sha1(path.encode("utf-8")).hexdigest(), "error": None}
```

Ruff (0.15.20)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)